src/platform/backends/shared/linux/process_factory.cpp

```diff
@@ -90,7 +99,16 @@ std::unique_ptr<mp::Process> mp::ProcessFactory::create_process(std::unique_ptr<
 {
     if (apparmor && !process_spec->apparmor_profile().isNull())
     {
-        return std::make_unique<AppArmoredProcess>(apparmor.value(), std::move(process_spec));
+        try
+        {
+            return std::make_unique<AppArmoredProcess>(apparmor.value(), std::move(process_spec));
+        }
+        catch (const mp::AppArmorException& e)
+        {
+            // TODO: This won't fly in strict mode (#1074), since we'll be confined by snapd
+            mpl::log(mpl::Level::warning, "apparmor", e.what());
+            return std::make_unique<BasicProcess>(std::move(process_spec));
```

Comment:
I am afraid the `process_spec` was already moved, it has a valid but indeterminate state at the this point. Not what we want to create a process with.

Reply:
Ah yes, /me really rushed this and didn't think through…